cumulusci/tasks/github/public_repo.py

```diff
@@ -0,0 +1,148 @@
+import os
+import shutil
+from datetime import datetime
+import github3.exceptions
+
+from cumulusci.core.exceptions import GithubException
+from cumulusci.core.github import get_github_api_for_repo
+from cumulusci.core.utils import process_bool_arg
+from cumulusci.core.utils import process_list_arg
+from cumulusci.tasks.github.base import BaseGithubTask
+from cumulusci.tasks.github.util import CommitDir
+from cumulusci.utils import cd
+from cumulusci.utils import download_extract_github_from_repo
+from cumulusci.utils import temporary_dir
+
+
+class PublishRelease(BaseGithubTask):
+    task_options = {
+        "repo_url": {"description": "The url to the public repo", "required": True,},
+        "branch": {
+            "description": "The branch to update in the target repo",
+            "required": True,
+        },
+        "version": {
+            "description": "The version number to release.  Also supports latest and latest_beta to look up the latest releases.",
+            "required": True,
+        },
+        "exclude": {"description": "A list of paths from repo root to exclude",},
+        "release_body": {
+            "description": "If True, the entire release body will be published to the public repo.  Defaults to False",
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+        self.options["exclude"] = process_list_arg(self.options.get("exclude", []))
+        self.options["release_body"] = process_bool_arg(
+            self.options.get("release_body", False)
+        )
+        if self.options["version"] == "latest":
+            self.options["version"] = str(self.project_config.get_latest_version())
+        elif self.options["version"] == "latest_beta":
+            self.options["version"] = str(
+                self.project_config.get_latest_version(beta=True)
+            )
+
+    def _get_target_repo_api(self):
+        gh = self.project_config.get_github_api(
+            self._target_repo_info["owner"], self._target_repo_info["name"],
+        )
+        return gh.repository(
+            self._target_repo_info["owner"], self._target_repo_info["name"],
+        )
+
+    def _run_task(self):
+        self._target_repo_info = self.project_config._split_repo_url(
+            self.options["repo_url"]
+        )
+        self.target_repo = self._get_target_repo_api()
+
+        with temporary_dir(chdir=False) as target:
```

Comment:
In Python 3 you can now just use tempfile.TemporaryDirectory as a context manager, if you don't want to cd to it.